aopy/analysis/base.py

```diff
@@ -1810,4 +1811,95 @@ def calc_confidence_interval_overlap(CI1, CI2):
+    weights = np.zeros((ntime-lags, nlabels, (1+lags)*nch, nfolds))*np.nan # (ntime, ntargets, nfeatures, nfolds)
+    cm = np.zeros((ntime-lags, nlabels, nlabels, nfolds))*np.nan
+    nwind = data.shape[0] - lags
+    for iwind in range(nwind):
+        lda = LinearDiscriminantAnalysis(solver='eigen', shrinkage=regularization)
+        end_wind = iwind+lags+1
+
+        # If there are nans at this timestep, return nan in all places
+        if np.sum(np.isnan(data[iwind:end_wind,:,:])):
+            if nfolds < 2:
+                decoding_accuracy[iwind] = np.nan
+                weights[iwind,:,:] = np.nan
+                cm[iwind,:,:] = np.nan
+            else:
+                decoding_accuracy[iwind,:] = np.nan
+                if return_weights:
+                    weights[iwind,:,:,:] = np.nan
+                if return_confusion_matrix:
+                    cm[iwind,:,:,:] = np.nan
+
+        else:
+            # If not cross validated
+            if nfolds < 2:
+                lda.fit(np.vstack(data[iwind:end_wind,:,:]).T - np.mean(np.vstack(data[iwind:end_wind,:,:]).T, axis=0), labels)
+                decoding_accuracy[iwind] = lda.score(np.vstack(data[iwind:end_wind,:,:]).T - np.mean(np.vstack(data[iwind:end_wind,:,:]).T, axis=0), labels)
+                predicted_labels = lda.predict(np.vstack(data[iwind:end_wind,:,:]).T - np.mean(np.vstack(data[iwind:end_wind,:,:]).T, axis=0))
+                cm[iwind,:,:,0] = confusion_matrix(labels, predicted_labels)
+                                                        
+            # if cross validated: 
+            else:
+                kf = model_selection.KFold(n_splits=nfolds,shuffle=True,random_state=None)
+                for ifold, (train_idx, test_idx) in enumerate(kf.split(np.vstack(data[iwind:end_wind,:,:]).T)):
+                    # print(ifold, train_idx, test_idx, iwind, end_wind)
+                    lda.fit(np.vstack(data[iwind:end_wind,:,train_idx]).T - np.mean(np.vstack(data[iwind:end_wind,:,train_idx]).T, axis=0), labels[train_idx])
+                    decoding_accuracy[iwind, ifold] = lda.score(np.vstack(data[iwind:end_wind,:,test_idx]).T - np.mean(np.vstack(data[iwind:end_wind,:,test_idx]).T, axis=0), labels[test_idx])
+                    predicted_labels = lda.predict(np.vstack(data[iwind:end_wind,:,test_idx]).T - np.mean(np.vstack(data[iwind:end_wind,:,test_idx]).T, axis=0))
+                    if return_weights:
+                        weights[iwind,:,:,ifold] = lda.coef_
+                    if return_confusion_matrix:
+                        cm[iwind,:,:,ifold] = confusion_matrix(labels[test_idx], predicted_labels, normalize='true')
+            
+    if return_weights & return_confusion_matrix:
+        return decoding_accuracy, weights, cm
+    elif return_weights: 
+        return decoding_accuracy, weights
+    elif return_confusion_matrix:
+        return decoding_accuracy, cm
+    else:
+        return decoding_accuracy
```

Comment:
To me. adding the return value for time axis that describes all time points for decoding is helpful. Maybe the center or right edge of each window is good?